src/streams.h

```diff
@@ -302,18 +302,16 @@ class CDataStream
 
         // Read from the beginning of the buffer
         unsigned int nReadPosNext = nReadPos + nSize;
-        if (nReadPosNext >= vch.size())
+        if (nReadPosNext > vch.size()) {
+            throw std::ios_base::failure("CDataStream::read(): end of data");
+        }
+        memcpy(pch, &vch[nReadPos], nSize);        
```

Comment:
```suggestion
        memcpy(pch, &vch[nReadPos], nSize);
```
Linter fails because of this. This is fixed later in btc 13753 together with lots of other cases.